src/builder.rs

```diff
@@ -60,7 +71,7 @@ pub enum OptionsError {
     MqttError(#[from] rumqttc::ClientError),
```

Comment:
We should better re-arenge the errors that are relative to the appropriate client error.

Reply:
I've re-arranged and refactored errors.